x-pack/metricbeat/module/aws/ec2/_meta/data.json

```diff
@@ -7,28 +7,32 @@
     "aws": {
         "ec2": {
             "cpu": {
+                "credit_balance": 172.95243,
+                "credit_usage": 0.001657,
+                "surplus_credit_balance": 0,
+                "surplus_credits_charged": 0,
                 "total": {
-                    "pct": 3.34375
+                    "pct": 0.066120218579236
                 }
             },
             "diskio": {
                 "read": {
                     "bytes": 0,
-                    "ops": 0
+                    "count": 0
                 },
                 "write": {
-                    "bytes": 14036172.8,
-                    "ops": 1320.4
+                    "bytes": 0,
+                    "count": 0
                 }
             },
             "network": {
                 "in": {
-                    "bytes": 112.8,
-                    "packets": 2
+                    "bytes": 325.6,
+                    "packets": 3.4
                 },
                 "out": {
-                    "bytes": 84.8,
-                    "packets": 2
+                    "bytes": 289.8,
```

Comment:
I'm just seeing this now but it's kind of odd that we have partial bytes?

Reply:
@ruflin Yeah good point. It's because these results are actually an `average` in last 5 minutes(with default monitoring) https://github.com/elastic/beats/blob/master/x-pack/metricbeat/module/aws/ec2/ec2.go#L354. `average` is the default statistic method used by cloudwatch. Other statistics can be used in cloudwatch are `Minimum`, `Maximum`, `Sum`, `Sample Count` and `percentile`. Maybe it's better to make this statistic method configurable here?

Reply:
thinking about this, should I add something like statistics=["average", "sum"] similar to cpu.metrics=["percentages", "ticks"] to allow users to apply more than one statistic methods?

Reply:
Just a low-hanging fruit idea, in the [naming conventions](https://www.elastic.co/guide/en/beats/devguide/current/event-conventions.html) we also have `avg` so you can also leave it like `out.bytes.avg`

I'm also curious about how Kibana might react if we set an index pattern with a floating point bytes field.

Reply:
Do we even have the counter value here or only the average?

Reply:
@ruflin counter value means the individual data points? We don't have the individual data points, we only can query the average/sum/min/max/... results from cloudwatch here.

Reply:
@sayden Thanks! Yeah I should add `avg` to the name but then it's going further away from mapping ec2 metrics to system metrics :-) Kibana didn't complain at all with the dashboard I set up for ec2 overview hmm.

Reply:
By counter I mean a value that keeps constantly increasing. This could be the sum here? Or is the sum also reset after 300s?

Reply:
@ruflin gotcha, no it's not a counter. It's a gauge value which only reflect average in each 300s.